webpack.config.ts

```diff
@@ -1,32 +1,56 @@
 import path from "path";
-import {Configuration, DefinePlugin} from "webpack";
+import {Configuration, DefinePlugin, EnvironmentPlugin, WebpackOptionsNormalized} from "webpack";
 
-module.exports = (env, options): Configuration => ({
-	devtool: options.mode !== 'production' ? 'source-map' : undefined,
-	context: path.join(__dirname, '/extension'),
+import config from "./mv3-hot-reload.config";
+
+const isDev = (options: WebpackOptionsNormalized) => options.mode !== "production";
+
+const srcDir = path.join(__dirname, "extension");
+const tsSrcDir = path.join(srcDir, "ts");
+const getEntry = (name: string) => {
+	return [path.join(tsSrcDir, name), ...(isDev ? [`mv3-hot-reload/${name}`] : [])];
+};
+
+module.exports = (_env: any, options: WebpackOptionsNormalized): Configuration => ({
+	devtool: isDev(options) ? "source-map" : undefined,
+	// @ts-ignore
+	devServer: {
+		hot: true,
+		firewall: false,
+		port: 3912,
+		devMiddleware: {
+			writeToDisk: true,
+		},
+		static: {
+			watch: false,
+		},
+		client: {
+			host: "localhost",
+		},
+	},
 	entry: {
-		'bundle': './ts/main/index.ts',
-		'options': './ts/options/index.ts',
-		'background': './ts/background/delegator.ts',
+		options: path.join(tsSrcDir, "options"),
+		bundle: getEntry("content"),
```

Comment:
wait why can't this be `"main"`?

Reply:
```ts
[watch:*src] ERROR in bundle
[watch:*src] Module not found: Error: Can't resolve 'mv3-hot-reload/main' in '/Users/aymendirar/Developer/library-thing'
[watch:*src] resolve 'mv3-hot-reload/main' in '/Users/aymendirar/Developer/library-thing'
[watch:*src]   Parsed request is a module
[watch:*src]   using description file: /Users/aymendirar/Developer/library-thing/package.json (relative path: .)
[watch:*src]     Field 'browser' doesn't contain a valid alias configuration
[watch:*src]     resolve as module
[watch:*src]       looking for modules in /Users/aymendirar/Developer/library-thing/node_modules
[watch:*src]         existing directory /Users/aymendirar/Developer/library-thing/node_modules/mv3-hot-reload
[watch:*src]           using description file: /Users/aymendirar/Developer/library-thing/node_modules/mv3-hot-reload/package.json (relative path: .)
[watch:*src]             using description file: /Users/aymendirar/Developer/library-thing/node_modules/mv3-hot-reload/package.json (relative path: ./main)
[watch:*src]               no extension
[watch:*src]                 Field 'browser' doesn't contain a valid alias configuration
[watch:*src]                 /Users/aymendirar/Developer/library-thing/node_modules/mv3-hot-reload/main doesn't exist
[watch:*src]               .tsx
[watch:*src]                 Field 'browser' doesn't contain a valid alias configuration
[watch:*src]                 /Users/aymendirar/Developer/library-thing/node_modules/mv3-hot-reload/main.tsx doesn't exist
[watch:*src]               .ts
[watch:*src]                 Field 'browser' doesn't contain a valid alias configuration
[watch:*src]                 /Users/aymendirar/Developer/library-thing/node_modules/mv3-hot-reload/main.ts doesn't exist
[watch:*src]               .js
[watch:*src]                 Field 'browser' doesn't contain a valid alias configuration
[watch:*src]                 /Users/aymendirar/Developer/library-thing/node_modules/mv3-hot-reload/main.js doesn't exist
[watch:*src]               as directory
[watch:*src]                 /Users/aymendirar/Developer/library-thing/node_modules/mv3-hot-reload/main doesn't exist
[watch:*src]       /Users/aymendirar/Developer/node_modules doesn't exist or is not a directory
[watch:*src]       /Users/aymendirar/node_modules doesn't exist or is not a directory
[watch:*src]       /Users/node_modules doesn't exist or is not a directory
[watch:*src]       /node_modules doesn't exist or is not a directory
[watch:*src] 
[watch:*src] webpack 5.74.0 compiled with 1 error in 3442 ms
```

it hates being called `main` -- i think they assume it's called `common` to do whatever they're doing and renaming it fixed it. it also kinda coincides with what is -- content scripts vs background stuff